.github/workflows/test.yml

```diff
@@ -58,5 +58,16 @@ jobs:
         run: |
           uv sync --extra dev
 
+      - name: Write halmos.toml
+        shell: bash
+        run: |
+          uv run python -m halmos.config \
+            | sed 's|storage-layout = "[^"]*"|storage-layout = "${{ matrix.storage-layout }}"|' \
+            | sed 's|cache-solver = [^[:space:]]*|cache-solver = ${{ matrix.cache-solver }}|' \
+            | tee tests/regression/halmos.toml
+
       - name: Run pytest
-        run: uv run pytest -v -k "not long and not ffi" --ignore=tests/lib --halmos-options="-st --disable-gc --solver-threads 1 --storage-layout ${{ matrix.storage-layout }} ${{ matrix.cache-solver }} --solver-timeout-assertion 0 ${{ inputs.halmos-options }}" ${{ inputs.pytest-options }}
+        shell: bash
+        run: |
+          uv run pytest -v -k "not long and not ffi" --ignore=tests/lib \
+            --halmos-options="-st --disable-gc --solver-threads 1 ${{ inputs.halmos-options }}" ${{ inputs.pytest-options }}
```

Comment:
cc @daejunpark -- instead of passing command line options (high pri), I pass the test config in halmos.toml (low pri), so that it can be customized with code annotations

Reply:
good idea! btw, can we just pass the options to halmos.config?
```
uv run python -m halmos.config --storage-layout ${{ matrix.storage-layout }} ${{ matrix.cache-solver }} | tee tests/regression/halmos.toml
```

also, can we also add --solver-timeout-assertion to halmos.config, instead of 3ecfa8f4aaf0e1770272f3b483a04c1726449da7?  ci performance fluctuates, and it's more robust to remove timeout limits by default.